python/ray/experimental/internal_kv.py

```diff
@@ -25,35 +25,64 @@ def _initialize_internal_kv(gcs_client: GcsClient):
     _initialized = True
 
 
+__NS_START_CHAR = b"@"
+
+
+def __make_key(namespace: Optional[str], key: bytes) -> bytes:
+    if namespace is None:
+        assert not key.startswith(__NS_START_CHAR)
+        return key
+    assert isinstance(namespace, str)
+    assert isinstance(key, bytes)
+    return b":".join([__NS_START_CHAR, namespace.encode(), key])
+
+
+def __parse_key(key: bytes) -> Tuple[Optional[str], bytes]:
+    assert isinstance(key, bytes)
+    if not key.startswith(__NS_START_CHAR + ":"):
+        return (None, key)
+
+    _, ns, key = key.split(b":", 2)
+    return (ns.decode(), key)
+
+
 @client_mode_hook(auto_init=False)
 def _internal_kv_initialized():
     return global_gcs_client is not None
 
 
 @client_mode_hook(auto_init=False)
-def _internal_kv_get(key: Union[str, bytes]) -> bytes:
+def _internal_kv_get(key: Union[str, bytes],
```

Comment:
I personally think it is better to have no leading underscore in the function name here, and one leading underscore for the make_key / parse_key functions. But it is ok to leave them as they are.

Reply:
it's hard to change the api right now. i'd prefer doing it later if we have time. the issue here is that, it's not just used by ray, but also other services. another reason is that ray client has proxy to this function.
this means we have to  1) keep both versions; 2) also make ray client use both versions. 3) update internal usage; 4) give time to let user update their usage.
so i'd prefer keeping it as now.

Reply:
Sounds good.